browser-extension/background.js

```diff
@@ -53,14 +62,60 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
       return []
     };
 
+    // Compute the number of videos to load in each category
+    const recentVideoToLoad = Math.round(request.videosNumber*oversamplingRatioForRecentVideos*recentVideoProportion);
+    const oldVideoToLoad = Math.round(request.videosNumber*oversamplingRatioForOldVideos*(1-recentVideoProportion));
+    const recentAdditionalVideoToLoad = 
+      Math.round(request.additionalVideosNumber*oversamplingRatioForRecentVideos*recentVideoProportionForAdditionalVideos);
+    const oldAdditionalVideoToLoad = 
+      Math.round(request.additionalVideosNumber*oversamplingRatioForOldVideos*(1-recentVideoProportionForAdditionalVideos));
+
     const process = async () => {
       const threeWeeksAgo = getDateThreeWeeksAgo()
-      const recent = await request_recommendations(`date_gte=${threeWeeksAgo}&language=${request.language}&limit=10`);
-      const old = await request_recommendations(`date_lte=${threeWeeksAgo}&language=${request.language}&limit=50`);
-      const recent_sub = getRandomSubarray(recent, 3);
-      const old_sub = getRandomSubarray(old, 4 - recent_sub.length);
-      const videos = getRandomSubarray([...old_sub, ...recent_sub], 4);
-      return { data: videos };
+
+      // Only one request for both videos and additional videos
+      const recent = await request_recommendations(
+        `date_gte=${threeWeeksAgo}&language=${request.language}&limit=${recentVideoToLoad+recentAdditionalVideoToLoad}`
+      );
+      const old = await request_recommendations(
+        `date_lte=${threeWeeksAgo}&language=${request.language}&limit=${oldVideoToLoad+oldAdditionalVideoToLoad}`
+      );
+      
+      // Cut the response into the part for the videos and the one for the additional videos
+      const videoRecent = recent.slice(0,recentVideoToLoad);
+      const videoOld = old.slice(0,oldVideoToLoad);
+      const additionalVideoRecent = recent.slice(recentVideoToLoad);
+      const additionalVideoOld = old.slice(oldVideoToLoad);
```

Comment:
That is not working as expected. The `recent` and `old` returned by the db are sorted so additional videos will always be taken from the bottom two thirds of the lists. Did you intend this?

I would propose:

```JavaScript
const numberOfRecentVideoToRespond = Math.round(request.videosNumber*recentVideoProportion);
const numberOfRecentAdditionalVideoToRespond = Math.round(request.additionalVideosNumber*recentVideoProportionForAdditionalVideos);

// fetch for both normal and additional videos
const recentVideos = getRandomSubarray(recent, numberOfRecentVideoToRespond + numberOfRecentAdditionalVideoToRespond)
// split the two lists
const recentVideos = recentVideos.slice(0, numberOfRecentVideoToRespond)
const additionalRecentVideos = recentVideos.slice(numberOfRecentVideoToRespond)

// TODO:
// - Compute number of required old and additionalOld videos based on lengths of lists obtained for recent and recentAdditional
// - Get the lists of videos with the same method from `old`
```

Reply:
I noticed that, but I believe it makes more sense like that. Most users will probably never click on the expand button, so I believe that it is crucial that this feature doesn't decrease the quality of the main recommendations. I believe that the current state of the branch doesn't change the main recommendations, and I think it is better if it remains so.

Reply:
Ok! I'm convinced by your reasonning. Let's stay with your proposition for now and reassess it later if we feel the need or want to change more things. Thanks!